lib/travis/build/script/crystal.rb

```diff
@@ -56,6 +58,18 @@ def script
           sh.cmd "crystal spec"
         end
 
+        def setup_cache
+          if data.cache?(:shards) && !cache_dirs.empty?
+            sh.fold 'cache.shards' do
+              directory_cache.add cache_dirs
+            end
+          end
+        end
+
+        def cache_slug
+          super << '-crystal'
```

Comment:
You'd probably want to add a piece about the crystal version in `cache_slug`, in order to avoid name collision.

Reply:
The crystal version, from the configuration, is nightly or latest. I could interpolate the configuration value in the slug. But if possible I would rather extract the value of `crystal --version` and use that in part of the slugh. Is there a way to do that? All other build scripts seems to emit the configuration value directly.